cmake/parthenonConfig.cmake.in

```diff
@@ -11,16 +11,26 @@
 # the public, perform publicly and display publicly, and to permit others to do so.
 #=========================================================================================
 
-# Favor using the kokkos package that was built with parthenon, if one has not been specified 
-if ( NOT Kokkos_FOUND )
+# Favor using the kokkos package that was built with parthenon, if one has not been specified
+if (NOT @EXTERNAL_KOKKOS@)
+  find_package(Kokkos 3 REQUIRED PATHS @EXTERNAL_KOKKOS@ NO_DEFAULT_PATH)
+else()
   find_package(Kokkos 3 REQUIRED PATHS @CMAKE_INSTALL_PREFIX@)
-  set(Kokkos_BUILT_WITH_CUDA @Kokkos_ENABLE_CUDA@)
-  if(${Kokkos_BUILT_WITH_CUDA})
-    string(FIND ${CMAKE_CXX_COMPILER} "nvcc_wrapper" FOUND_WRAPPER)
-    if( ${FOUND_WRAPPER} EQUAL "-1" ) 
+endif()
+
+set(Kokkos_BUILT_WITH_CUDA @Kokkos_ENABLE_CUDA@)
+if(${Kokkos_BUILT_WITH_CUDA})
+  string(FIND ${CMAKE_CXX_COMPILER} "nvcc_wrapper" FOUND_WRAPPER)
+  if( ${FOUND_WRAPPER} EQUAL "-1" )
+    if (NOT @EXTERNAL_KOKKOS@)
       message(WARNING "Kokkos was built with cuda, recommend setting CMAKE_CXX_COMPILER to @CMAKE_INSTALL_PREFIX@/bin/nvcc_wrapper")
+    else()
+      message(WARNING "Kokkos was built with cuda, recommend setting CMAKE_CXX_COMPILER to @EXTERNAL_KOKKOS@/bin/nvcc_wrapper")
     endif()
   endif()
 endif()
 
+find_package(MPI COMPONENTS CXX)
+find_package(OpenMP COMPONENTS CXX)
```

Comment:
Ideally, we could use "PATHS" or "HINTS" to indicate where find_package should search, this would help prevent finding the wrong Config file. There has to be a way to get the location of the config files that are actually being used... though the documentation so far has been less than forthcoming.

Reply:
You might try using 

find_package(MPI COMPONENTS CXX PATHS @MPI_CONFIG_DIR@) 

Though this might only be defined if CMakePackageConfigHelpers was used when MPI was installed. 
https://cmake.org/cmake/help/v3.4/module/CMakePackageConfigHelpers.html

The same might be true for the other thirdparty applications.

Reply:
For `OpenMP`, it's actually the compiler that provides the package, so it's entangled with the compiler itself. Really you'd have to check if the same compiler was used for both Parthenon and the app.

I'd say it's not worth spending your time doing this validation. It's a lot of engineering effort to validate all the things you'd need to check to ensure there's compatibility between the app and the dependency, and I think for little value because I think people are generally accustomed to the idea that they need to compile their dependencies with the same compiler and libraries as with their app.